pkg/prometheus/promcfg.go

```diff
@@ -1137,6 +1153,19 @@ func getLimit(user uint64, enforced *uint64) uint64 {
 	return user
 }
 
+func addressSharding(relabelings []yaml.MapSlice, shards int32) []yaml.MapSlice {
```

Comment:
Nit: Seems like a confusing function name to me, had to look at the function to understand what it did, maybe something more self-descriptive like `injectSharding` or `generateSharding` wdyt?

Would a unit test be good here for this function to cover some possible edge cases, wdyt?

Reply:
Agreed, something with "generate" in the function name would make more sense.

Reply:
There are not really any edge cases here though that aren't covered elsewhere already, as this is really only text templating.

Reply:
Renaming to `generateAddressShardingRelabelingRules`